### PR TITLE
Support scheme v3.1

### PR DESCRIPTION
**Pull request by jmatsu**
v3.1 introduces target sdk ability so this PR adds interfaces to return signature information with target sdk.

- I've deprecated many methods that rely on non-rotated signing.
- Revised test assets because some of them are not properly configured

### CODE REVIEW

File: lib/android_apk/apksigner.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      if lines[0].include?("minSdkVersion=")
+        lines.each do |line|
+          next if (m = line.match(TARGET_SDK_PART_REGEX)).nil?
+
+          sdk_versions = [m[1], m[2]]
+          signers[sdk_versions] ||= []
+          signers[sdk_versions].push(line)
+        end
+      else
+        # TODO: support multiple signers
```
Comment (jmatsu) [inline lib/android_apk/apksigner.rb:88]: Multiple key signing is not supported by Google Play or the general usecase of App Bundle, so this has been to TODOs.

File: lib/android_apk.rb [hunk 2]
```diff
@@ -67,9 +71,9 @@ class AndroidApk
   # @return [String] Return Integer string which is defined in AndroidManifest.xml
   attr_accessor :target_sdk_version
 
-  # The trusted signature lineage. The first element is the same to the signing signature of the apk file.
-  # @return [Array<String>] empty if it's unsigned.
-  attr_reader :trusted_signature_lineage
+  # An object contains lineages and certificate fingerprints
+  # @return [AndroidApk::AppSignature] a signature representation
+  attr_reader :app_signature
```
Comment (jmatsu) [inline lib/android_apk.rb:76]: This exposes v1, v2, v3, v3.1 compatible interfaces.

File: lib/android_apk/app_signature.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        end
+
+        # rollback is a special case
+        if !(rollback_index = lineages_from.index { |lineage| lineage[digest_method] == target_app_signing_digest }).nil? && lineages_from[rollback_index]["rollback"]
+          return fingerprint_to.slice(::AndroidApk::SignatureDigest::MD5, ::AndroidApk::SignatureDigest::SHA1, ::AndroidApk::SignatureDigest::SHA256)
+        end
+
+        return nil if sdk_version < V3_SCHEME_SDK_INT
+
+        # Return true if the target apk contains the current signing signature.
```
Comment (jmatsu) [inline lib/android_apk/app_signature.rb:53]: These verification are only for v3. i.e. key-rotation

**Review by satsukies — APPROVED** [review]
LGTM

**Pull request merged**